action.yml

```diff
@@ -138,14 +218,24 @@ runs:
           )
       shell: python
 
-    - name: Push XML report to ${{ inputs.diff-storage }}
-      if: contains(inputs.diff, 'true') && (steps.branch-names.outputs.current_branch == inputs.diff-branch)
-      uses: JamesIves/github-pages-deploy-action@v4.2.2
-      with:
-        branch: ${{ inputs.diff-storage }}
-        folder: coverage-report-action
-        token: ${{ github.token }}
-        commit-message: 'Coverage report for ${{ github.sha }}'
+    - name: Fail if coverage worsened
+      run: |
+        if [[ "${{ inputs.coverage-reduction-failure }}" == "true" ]]
+        then {
+          if [[ ${{ env.pycobertura_status }} == "2" ]]
+          then {
+            echo "Code changes worsened the overall coverage."
+            exit 1
+          }
+          elif [[ ${{ env.pycobertura_status }} == "3" ]]
```

Comment:
We have to decide if we want to fail in this case. Additionally, we can consider to fail when `status = 1`.
We can also introduce a new parameter to control this.
Quoting the [documentation](https://github.com/aconrad/pycobertura):
> Upon exit, the diff command may return various exit codes:
> 0: all changes are covered, no new uncovered statements have been introduced
> 1: some exception occurred (likely due to inappropriate usage or a bug in pycobertura)
> 2: the changes worsened the overall coverage
> 3: the changes introduced uncovered statements but the overall coverage is still better than before